src/websocket/atomic-swap.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import BaseWebSocket, { WsOptions } from './base';
+
+/**
+ * Handles websocket connections and message transmission
+ *
+ * This class extends the base websocket class.
+ *
+ * @class
+ * @name AtomicSwapWebSocket
+ */
+class AtomicSwapWebSocket extends BaseWebSocket {
+  constructor(options: WsOptions) {
+    super(options);
+  }
+
+  /**
+   * Handle message receiving from websocket
+   *
+   * @param {Object} evt Event that has data (evt.data) sent in the websocket
+   */
+  onMessage(evt) {
+    const message = JSON.parse(evt.data)
+    const _type = message.type.split(':')[0]
```

Comment:
Why are we splitting the type with `:`?

Reply:
It wasn't necessary. Removed on [d1d252f](https://github.com/HathorNetwork/hathor-wallet-lib/pull/533/commits/d1d252f142c486672c57747daa09e63a54d144b2)